app/helpers/authorization.rb

```diff
@@ -0,0 +1,32 @@
+module Authorization
+
+	def authorize_or_redirect
+		redirect '/login' unless session[:user_id]
+		permissions = User.find(session[:user_id]).access_level
+		if permissions == "superadmin"
+		 	if user_id_from_params.is_a? Integer
+				redirect "/users/#{session[:user_id]}" unless user_id_from_params == session[:user_id]
+			end
+		else
+			check_member_permissions
+		end
+	end
+
+	def user_id_from_params
```

Comment:
This is a little bit crazy. If you want a param use sinatras build in param tools. You should not need to regex your own params. There be dragons.